pkg/gadget-service/instance-manager/file-store/filestore.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2023-2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filestore
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/moby/moby/pkg/namesgenerator"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/protobuf/encoding/protojson"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+	instancemanager "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/instance-manager"
+)
+
+var GadgetFilePath = "/var/lib/ig"
+
+type FileStore struct {
+	api.GadgetInstanceManagerServer
+	instanceMgr *instancemanager.Manager
+}
+
+func New(mgr *instancemanager.Manager) (instancemanager.Store, error) {
+	fs := &FileStore{
+		instanceMgr: mgr,
+	}
+	err := fs.init()
+	if err != nil {
+		return nil, err
+	}
+	return fs, nil
+}
+
+// init scans GadgetFilePath for existing gadget configurations and runs them.
+func (s *FileStore) init() error {
+	// Load & Apply
+	err := os.MkdirAll(GadgetFilePath, 0o744)
```

Comment:
> Any reason not to pick the same permissions here?

Nope, it should be aligned, I will change it. This code is older than the oci store, hehe. But if we change it to somewhere in `/etc/ig` like you suggested, which value would you recommend there?

Reply:
If it is just configuration without credentials, 744 for files and 755 for directories is ok. If it can contain credentials, 700 for both files and directories.

- `/var/lib/ig`: 755
  - `oci-store`: 700 because it can contain data pulled with `ig image pull` that were private
  - `gadgets`: 755 or 700, depending if it can contain credentials.

It's ok to have different permissions for different directories. But if `/var/lib/ig` needs to have different permissions than `/var/lib/ig/gadgets`, then it needs two separate `os.MkdirAll`.